### ISSUE TITLE
Unhandled Runtime Error in mantine-datatable: m.map is not a function

### ISSUE THREAD

**Issue by ecatugy** [icflorescu/mantine-datatable#631]
# Bug Report: Unhandled Runtime Error in Next.js Application

**Description:**

An unhandled runtime error is occurring in our Next.js application using React, TypeScript, and `mantine-datatable` version 7.11.2. The error suggests a problem with the `.map()` function being called on a non-array object.

**Environment:**

- **Next.js version:** 14.2.5
- **React:** 18.3.1
- **TypeScript:** 5.5.4
- **mantine-datatable:** 7.11.2

**Unhandled Runtime Error:**
**TypeError: m.map is not a function**

**Comment by icflorescu** [comment]
The provided info is too vague.
Please provide more context (such as a codesandbox/repo reproducing the issue).

**Comment by bastiengrignon** [comment]
I've came accross this error too but it happens when you don't set the `records` or `columns` so it seems logic to fail if you don't provide anything for your datatable

**Comment by gilank017** [comment]
can you help me to solve this problem ? 

<img width="1675" height="968" alt="Image" src="https://github.com/user-attachments/assets/10bc6f4a-9e5d-4357-b5d7-ae8183be890f" />

this is my source code

<img width="852" height="725" alt="Image" src="https://github.com/user-attachments/assets/2f22e561-a16f-4bed-9f7f-a53572fde1d4" />

**Comment by bastiengrignon** [comment]
What's your version of mantine-datatable ? I'm using `v7.17.1` (the latest for v7) on a project and `v8.2.0` (the latest for v8) on another project and both works fine.
Have you tried re-installing it with your package manager ?

**Comment by chatne** [comment]
I had this issue and I used some time in the debugger to figure this out.

**TLDR; column order is stored in local storage with default key of empty string. If there is already unrelated data in the storage with that key the datatable can crash.**

Fix by specifying `storeColumnsKey` or by clearing the value from the storage.

---

When `<DataTable />` is initialized without specifying `storeColumnsKey` the default is used and for some reason the default is just empty string.

https://github.com/icflorescu/mantine-datatable/blob/28ca8473690bd11cac58c9a2137b0dbd0e17a06b/package/hooks/useDataTableColumnReorder.ts#L51-L55

I happened to have a string value in the local storage with key as empty string. That value ended up in the `columnsOrder` variable and then the first `.map`-call causes the crash.

https://github.com/icflorescu/mantine-datatable/blob/28ca8473690bd11cac58c9a2137b0dbd0e17a06b/package/hooks/useDataTableColumns.ts#L84-L85

This seems pretty fragile design to me. I don't even use column order but the storage is still used regardless by `dragToggle` in a provider in the constructor.

https://github.com/icflorescu/mantine-datatable/blob/28ca8473690bd11cac58c9a2137b0dbd0e17a06b/package/DataTable.tsx#L155-L161
https://github.com/icflorescu/mantine-datatable/blob/28ca8473690bd11cac58c9a2137b0dbd0e17a06b/package/DataTable.tsx#L253-L255

Maybe the key should use a safer default? Or just require specifying it for when the storage is used?
